code/game/objects/structures/cannons/mounted_guns/mounted_gun.dm

```diff
@@ -187,3 +228,135 @@
 	throwforce = 0
 	w_class = WEIGHT_CLASS_BULKY
 	projectile_type = /obj/projectile/bullet/shrapnel
+
+/obj/structure/mounted_gun/ratvarian_repeater
+
+	name = "Ratvarian Repeater"
+	desc = "''Brains? Bronze? Why not both?''"
+	icon_state = "ratvarian_repeater"
+	icon_state_base = "ratvarian_repeater"
+	icon_state_fire = "ratvarian_repeater_fire"
+	loading_message = "gun charged"
+	anchored = FALSE
+	anchorable_gun = TRUE
+	uses_ammo = FALSE
+	has_loading_delay = TRUE
+	load_delay = 30
+	max_shots_per_fire = 12
+	shots_per_load = 12
+	shots_in_gun = 12
+	fire_sound = 'sound/items/weapons/thermalpistol.ogg'
+	last_fire_sound = 'sound/items/weapons/thermalpistol.ogg'
+	projectile_type = /obj/projectile/beam/laser/musket/repeater
+	loaded_gun = TRUE
+	fully_loaded_gun = TRUE
+	fire_delay = 1
+	shot_delay = 2
+	firing_shakes_camera = FALSE
+	custom_materials = list(
+	/datum/material/iron = SHEET_MATERIAL_AMOUNT * 5.25,
+	/datum/material/bronze = SHEET_MATERIAL_AMOUNT * 5,
+	/datum/material/glass = SHEET_MATERIAL_AMOUNT * 1.29)
+/obj/structure/mounted_gun/ratvarian_repeater/dump_contents() // for some god-forsaken reason trying to break this thing attempts to dump nonexistient contents and crashes the game. dont remove this.
```

Comment:
```suggestion

/obj/structure/mounted_gun/ratvarian_repeater/dump_contents()
	return
```

The reason it crashes is simply because you didn't implement this proc
```
/obj/proc/dump_contents()
	SHOULD_CALL_PARENT(FALSE)
	CRASH("Unimplemented.")
```
You should probably just find out why it's calling that proc instead and stop it from doing that.